src/app/search-bar/search-bar.component.ts

```diff
@@ -65,8 +65,15 @@ export class SearchBarComponent implements OnInit, AfterViewInit {
     let instantsearch = JSON.parse(localStorage.getItem('instantsearch'));
 
     if (instantsearch && instantsearch.value) {
-      this.store.dispatch(new queryactions.QueryServerAction({'query': event, start: this.searchdata.start, rows: this.searchdata.rows}));
+      this.store.dispatch(new queryactions.QueryServerAction({'query': event, start: 0, rows: this.searchdata.rows}));
       this.displayStatus = 'showbox';
+      this.hidebox(event);
+    } else {
+      if (event.which === 13) {
```

Comment:
when enter is pressed this code won't run since we have ngmodelchanges enabled and it doesn't send which paramter in the event.
change the onEnter funtion instead.